app/models/registration_period.rb

```diff
@@ -21,6 +21,10 @@ class RegistrationPeriod < ApplicationRecord
   validates :started_on, presence: { message: "Enter a start date" }
   validates :finished_on, presence: { message: "Enter an end date" }
 
+  def self.for_date(date)
+    where('range @> ?::date', date).first
```

Comment:
If you rebase to bring in the changes made in #776 this can be

```
find_by(*date_in_range(date))
```

Reply:
Also this feels like it's going to be useful - but it's not suitable for `Interval` because some `Period` types (mentors) can overlap.

I wonder if there should be another module for date functions common to exclusive intervals, `ExclusiveInterval` perhaps?

Not yet, just thinking.

Reply:
Ah missed that. Nice! I've updated to use it and renamed the method for clarity.